### PR TITLE
Add "WebKit" as a product

### PR DESCRIPTION
**Pull request by gsnedders**
This largely mimics what was done in #3161; let me know if I've missed anything.

### CODE REVIEW

**Comment by gsnedders** [comment]
Also: does it still require any work on the admin side to add things to the backend?

**Comment by gsnedders** [comment]
That said, we might want to call this "macOS WebKit" given how naming is going for Chrome (see also #3214).

**Comment by past** [comment]
The changes look good to me and I agree that "macOS WebKit" would be better. If you have sorted out the authentication stuff for uploading results, I don't think you need anything else on the backend side.

**Review by KyleJu — APPROVED** [review]
LGTM after addressing the naming!

**Comment by DanielRyanSmith** [comment]
It looks like there is a test that needs to be updated in `components/test/test-search.html`

**Comment by gsnedders** [comment]
Seemingly:

```js
    test('browser test status eq', () => {
      // All known browsers.
      for (const browser of AllBrowserNames) {
        assertQueryParse(browser + ':ok', {exists: [{product: browser, status: 'OK'}]});
      }
```

My guess is that `webapp/components/test-search.js` isn't coping with having one browser name be a prefix of another.

Does anyone still around here understand Ohm well enough to quickly know the solution here?

File: webapp/components/product-info.js [hunk 1]
```diff
@@ -45,7 +46,7 @@ const versionPatterns = Object.freeze({
 
 // The set of all browsers known to the wpt.fyi UI.
 const AllBrowserNames = Object.freeze(['android_webview', 'chrome_android', 'chrome_ios', 'chrome',
-  'deno', 'edge', 'firefox', 'flow', 'node.js', 'safari', 'servo', 'webkitgtk']);
+  'deno', 'edge', 'firefox', 'flow', 'node.js', 'safari', 'servo', 'webkit', 'webkitgtk']);
```
Comment (DanielRyanSmith) [inline webapp/components/product-info.js:49]:
```suggestion
  'deno', 'edge', 'firefox', 'flow', 'node.js', 'safari', 'servo', 'webkitgtk', 'webkit']);
```

I don't know why, but rearranging these made the test pass for me. Maybe someone smarter than me can figure out why. 😅

**Comment by KyleJu** [comment]
CI passed. Merging!

**Comment by gsnedders** [comment]
> LGTM after addressing the naming!

We never did address this! But it sounded like you were hoping to make progress on #3214 soonish?

**Comment by KyleJu** [comment]
> macOS WebKit


> We never did address this! But it sounded like you were hoping to make progress on #3214 soonish?

That is my bad! We do intend to address it in Q2, but the renaming should be done in this PR. Would you mind sending a follow-up PR?

**Pull request merged**